### PR TITLE
switched to globalconfig

### PR DESCRIPTION
**Pull request by ehlerst**
I have switched this to globalconfig however i am pretty new to all of this. I don’t know how to test this further. Any feedback on something else i should do for this merge is appreciated. 

It does seem to take the options and work which is what i'm after. I want to be able to more easily startup statsd in docker containers.

```
SD_ADMIN_ADDR=:9226 ./build/statsdaemon 
Profiling endpoint listening on ""
2017/05/13 15:04:58 statsdaemon instance 'MyServerIsbetterThanYours' starting
2017/05/13 15:04:58 listen tcp: missing port in address ""
Listening on :9226
2017/05/13 15:04:58 listening on :8125

```

### CODE REVIEW

**Comment by ehlerst** [comment]
I broke the filecheck, its starts with/without /etc/statsdaemon.ini in place. Ill look at it and try and update tonight.

**Comment by Dieterbe** [comment]
> I don’t know how to test this further. Any feedback on something else i should do for this merge is appreciated.

I think you're already doing the same as what I would do . just that you can set values via environment and via the config, verify that env overrides config, and if nothing specified, that it uses the defaults.

on first glance this looks OK but haven't looked in depth yet

**Comment by ehlerst** [comment]
hmm, do we want it to bomb out if the config file isn’t there? since everything is a flag now maybe it makes more sense to leave it like this. 

Fixed the help output. I didnt realize it already displayed the default values.
Output syntax of -h now.

```
./build/statsdaemon -h
Usage of ./build/statsdaemon:
  -admin_addr string
    	listener address for admin port (default ":8126")
  -config_file string
    	config file location (default "/etc/statsdaemon.ini")
  -cpuprofile string
    	write cpu profile to file
  -debug
    	log outgoing metrics, bad lines, and received admin commands
  -flush_counts
    	send count for counters (using prefix_counters)
  -flush_interval int
    	flush interval in seconds (default 10)
  -flush_rates
    	send count for counters (using prefix_counters) (default true)
  -graphite_addr string
    	graphite carbon-in url (default "127.0.0.1:2003")
  -instance string
    	instance name, defaults to short hostname if not set (default "$HOST")
  -legacy_namespace
    	legacy namespacing (not recommended) (default true)
  -listen_addr string
    	listener address for statsd, listens on UDP only (default ":8125")
  -max_timers_per_s uint
    	max timers per second (default 1000)
  -memprofile string
    	write memory profile to this file
  -percentile_thresholds string
    	percential thresholds (used by timers)
  -prefix_counters string
    	counters prefix (default "stats_counts.")
  -prefix_gauges string
    	gauges prefix (default "stats.gauges.")
  -prefix_m20_counters string
    	counters 2.0 prefix
  -prefix_m20_gauges string
    	gauges 2.0 prefix
  -prefix_m20_rates string
    	rates 2.0 prefix
  -prefix_m20_timers string
    	timers 2.0 prefix
  -prefix_rates string
    	rates prefix, it is recommended that you use stats.rates if possible (default "stats.")
  -prefix_timers string
    	timers prefix (default "stats.timers.")
  -processes int
    	number of processes to use (default 1)
  -profile_addr string
    	listener address for profiler
  -proftrigger_cpu_dur string
    	profiler cpu duration (default "5s")
  -proftrigger_cpu_freq string
    	profiler cpu frequency (default "60s")
  -proftrigger_cpu_min_diff string
    	profiler cpu min difference (default "1h")
  -proftrigger_cpu_thresh int
    	profiler cpu threshold (default 80)
  -proftrigger_heap_freq string
    	profiler heap frequency (default "60s")
  -proftrigger_heap_min_diff string
    	profiler heap min difference (default "1h")
  -proftrigger_heap_thresh int
    	profiler heap threshold (default 10000000)
  -proftrigger_path string
    	profiler file path (default "/tmp")
  -version
    	print version string
```

File: cmd/statsdaemon/main.go [hunk 1]
```diff
@@ -20,7 +20,7 @@ import (
 	"net/http"
 	_ "net/http/pprof"
 
-	"github.com/stvp/go-toml-config"
+	"github.com/rakyll/globalconf"
```
Comment (Dieterbe) [inline cmd/statsdaemon/main.go]: this library is not maintained anymore.
we use https://github.com/grafana/globalconf now

**Comment by Dieterbe** [comment]
> hmm, do we want it to bomb out if the config file isn’t there?

no, it is now perfectly legitimate to configure statsdaemon via env vars and not have a config file.

File: cmd/statsdaemon/main.go (hunk 1 above)
Comment (ehlerst) [inline cmd/statsdaemon/main.go]: Moved it to the grafana one. Also took out the vendor lock, assuming with it under grafana repo we want the latest always?

**Pull request merged**